### PR TITLE
Added recipe for dask-sql

### PR DESCRIPTION
**Pull request by nils-braun**
This PR adds a conda-recipe for [dask-sql](https://github.com/nils-braun/dask-sql).
It is basically "just" a simple python package from pipy, with the only exception that it also contains a java class that needs to be compiled with maven.

So far there has only been a first release candidate, no "official" release of the package.

This is my first PR to conda-forge - I am happy for any suggestions on how to improve!
Thanks for this great project.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/dask-sql/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/dask_sql-{{ version }}.tar.gz
+  sha256: 683e4c9759ef0a0d475aadc42d976125a04f9cc7d32065d11f48bd27282c46a1
+
+build:
+  noarch: python
```
Comment (scopatz) [inline recipes/dask-sql/meta.yaml:14]: Is this really noarch python, even with the Java bits?

Comment (nils-braun) [reply recipes/dask-sql/meta.yaml:14]: Thanks for the review @scopatz! 
I think "noarch" should be fine - at least this is what I expect from Java.
The "python" part also is not 100% clear to me, so thanks for asking. I understood the conda-forge documentation that there are no C++ compiled parts - which is true - but I do not know if also the java "compilation" counts here.
As it is basically a performance optimization on the build side I am also fine with removing the "python" part. Can/should I replace it with "noarch generic" then or remove this part completely?

Comment (mariusvniekerk) [reply recipes/dask-sql/meta.yaml:14]: Is the jar just a data-file part of the python package?

Should be fine to regard it as noarch

Comment (nils-braun) [reply recipes/dask-sql/meta.yaml:14]: The jar is created during the [python package installation](https://github.com/nils-braun/dask-sql/blob/main/setup.py#L38) via a [call](https://github.com/nils-braun/dask-sql/blob/main/setup.py#L9) to `mvn` and then specified in the `setup.py` as `package_data` [link](https://github.com/nils-braun/dask-sql/blob/main/setup.py#L62).
I honestly do not know if this is the best way to do it... 
But it is not already included in the repo, if this was your question.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**